src/Service/Form/Submission/FormSubmissionService.php

```diff
@@ -75,28 +83,69 @@ public function createDownload(FormSubmission $formSubmission): string
         return $tempFile;
     }
 
+    /**
+     * @param array<string> $formSubmissionIds
+     *
+     * @throws \Exception
+     */
+    public function createDownloadForMultiple(array $formSubmissionIds): string
```

Comment:
You should look at a solution for streaming a zip

Reply:
How about this library? https://github.com/maennchen/ZipStream-PHP

Reply:
OK, this implementation looks ok: https://stackoverflow.com/questions/44697218/using-zipstream-in-symfony-streamed-zip-download-will-not-decompress-using-arch